ASSIGNMENTS/slideInOnScroll/index.html

```diff
@@ -0,0 +1,195 @@
+
+
+
+
+  </div>
+
+  <script>
+    function debounce(func, wait = 20, immediate = true) {
+      var timeout;
+      return function() {
+        var context = this, args = arguments;
+        var later = function() {
+          timeout = null;
+          if (!immediate) func.apply(context, args);
+        };
+        var callNow = immediate && !timeout;
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+        if (callNow) func.apply(context, args);
+      };
+    }
+
+    //helper function to figure out if a given elemnet is entering the viewport
+    function isArrivingWithinWindow(element){
+      //
+      //console.log(element.offsetTop,' ',element.height)
+      //for half of the image appear within the viewpport
+      // the condition must be , mid-y of image must be less
+      //than window scroolY+window.innerHeight
+      const midYofElement = element.offsetTop+element.height/2;
+      if(midYofElement<window.scrollY ){
+        //console.log('moving out')
+        return true;
+      } else if( midYofElement< window.scrollY+ window.innerHeight){
+        //console.log("moving In")
+        return true;
+      } else{
+        //console.log('moving out')
+        return false;
+      }
+    }
+
+    const allImages = document.querySelectorAll('.slide-in');
+    console.assert(allImages!== null,'image elements must exists')
+   // console.log(allImages)
+
+   const totalPages = document.querySelector('.total-page');
+   const currentPages = document.querySelector('.current-page');
+
+    //We need to get called whenever user scrolls
+    window.addEventListener('scroll',(event)=>{
+      allImages.forEach((image)=>{
+        if(isArrivingWithinWindow(image)){
+          image.classList.add('active');
+        } else{
+          image.classList.remove('active');
+        }
+      })
+      
+    //console.log('total: ',document.body.offsetHeight)
+    //console.log('window: ', window.innerHeight)
+    //console.log('scroolY: ',window.scrollY)
+
+    totalPages.textContent = Math.round(document.body.offsetHeight/window.innerHeight) ;
```

Comment:
can move this to its own function to make it more readable. Also handle window resize case.